torch/compiler/__init__.py

```diff
@@ -0,0 +1,319 @@
+    recursively invoked functions within it. If not provided, a context manager will be returned.
+    - recursive (optional): A boolean value indicating whether the disabling should be recursive.
+    If set to True (default), compilation is completely skipped on the decorated function frame
+    and any recursively invoked functions within it. If set to False, compilation skips frames
+    associated with the function code but still processes recursively invoked frames.
+
+    Returns:
+    - If `recursive=True` and `fn` is provided, a decorated version of the function `fn` is returned,
+    with compilation disabled for the decorated function frame and any recursively invoked functions.
+    - If `recursive=True` and `fn` is not provided, a context manager is returned, allowing compilation
+    to be disabled within a specific code block.
+    - If `recursive=False`, the `skip()` function is returned, which allows compilation to skip frames
+    associated with the function code but still process recursively invoked frames.
+
+    Note:
+    - When using the decorator or context manager compilation processing is selectively disabled for
+    the decorated function frame and any recursive function calls, depending on the `recursive` flag.
+    - The function internally uses the `innermost_fn()` function to ensure that the innermost function
+    is decorated when `fn` is provided.
+    - The `skip()` function is used when `recursive=False` to skip frames associated with the function code
+    but still process recursively invoked frames.
+
+    Example:
+    1. Using the decorator with recursive disabling:
+
+    ::
+      @disable(recursive=True)
+      def my_function():
+
+    In this example, `my_function()` is decorated with compi disabled, meaning that compilations
+    processing will be skipped for the function frame and any recursive function calls within it.
+
+    2. Using the context manager with recursive disabling:
+
+    ::
+      with disable(recursive=True):
+
+    In this example, the code block within the `with` statement will have compilation disabled, meaning
+    that compilations processing will be skipped for the code within the block and any recursive function
+    calls within that code.
+
+    3. Using the skip function with non-recursive disabling:
+
+    ::
+      disable(recursive=False)(my_function)
+
+    In this example, `my_function()` is wrapped with the `skip()` function, which disables compilations
+    processing for the function frame but still processes recursively invoked functions.
+
+    """
+
+    return torch._dynamo.disable(fn, recursive)
+
+def list_options() -> Dict[str, Any]:
+    """
+    This function returns a dictionary that provides information about the available
+    optimizations and debug configurations that can be used with the `torch.compile()`
+    function. The options are documented in `torch._inductor.config`.
```

Comment:
IMO if this needs to reference an non-public file this function shouldn't be public.

Reply:
Hmm yeah can include this when we make the configs public